scripts/verify-knife

```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
+
+# Verify mmfile $1 (default "set.mm") and discouraged file $2 (default
+# "discouraged") using metamath-knife.
+# Usage:
+# verify-knife [source=set.mm] [discouraged_file=discouraged]
+
+source="${1:-set.mm}"
+discouraged_file="${2:-discouraged}"
+
+metamath-knife ${source} --verify \
+                         --verify-parse-stmt \
+                         --verify-usage \
+                         --verify-markup
+
+if [ $? != 0 ]; then
+  exit 1
+fi
+
+metamath-knife ${source} -D discouraged.new >/dev/null
```

Comment:
Why is this command guaranteed to always succeed, and has no return value check?  Even if the current implementation never fails, a future version might do.
Or use && to start the diff only if metamath-knife succeeds, for example
   metamath-knife ..params.. && diff ..params..

Reply:
It's not, the intent was that failures are caught by diff not finding discouraged.new. I updated it to use &&.